### PR TITLE
Use trash on windows when deleting environments

### PR DESCRIPTION
**Pull request by rmcgibbo**
@asmeurer, this also fixes the issue in #520. I watched the .trash folder, and it does get removed.

### CODE REVIEW

File: tests/test-recipes/metadata/jinja2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package:
+  name: conda-build-test-jinja2-in-recipe
+  version: 1.0
+
+build:
+  number: 0
+
+requirements:
+    build:
+        - jinja2
```
Comment (asmeurer) [inline tests/test-recipes/metadata/jinja2/meta.yaml]: You should add python here so that it builds against the same Python version as conda build.

**Comment by asmeurer** [comment]
I would like to get the trash stuff in rm_rf itself, but this will work for now.

**Comment by asmeurer** [comment]
I went ahead and just fixed the python thing myself.

**Comment by rmcgibbo** [comment]
Sweet.

**Comment by Korijn** [comment]
I just submitted issue #524 yesterday which seems closely related to #520. Do you think this PR will fix that issue as well? When will this PR be included in a release so that I can test?

**Comment by asmeurer** [comment]
Yes I think this will fix that as well.

**Comment by jasongrout** [comment]
Thanks for getting this in.  I was pulling my hair out trying to debug the problem with _speedups.pyd.

**Comment by jasongrout** [comment]
It looks like this might be the real solution to https://github.com/conda/conda-build/issues/211 too.

**Pull request merged**